### PR TITLE
For #8172: Add a new "Fenix" Leanplum attribute

### PR DESCRIPTION
**Pull request by ValentinTimisica**
### Pull Request checklist

- [x] **Tests**: This PR does not include thorough tests.
- [x] **Screenshots**: This PR does not include screenshots or GIFs.
- [x] **Accessibility**: The code in this PR does not include any user facing features.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by boek** [comment]
Can you please fill out a data review request? :) https://github.com/mozilla/data-review/blob/master/request.md

**Comment by ValentinTimisica** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
- If the current app is Fenix.

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- Leanplum marketing campaigns.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* N/A

4) **Can current instrumentation answer these questions?**
- No. For future leanplum campaigns we need new leanplum attributes

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

- Category 1.

6) **How long will this data be collected?**

- N/A

7) **What populations will you measure?**

- All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

- Leanplum

9) **Where do you intend to share the results of your analysis?**

- Only on Leanplum with the people who have the right permissions.

**Comment by ValentinTimisica** [comment]
As @lwa-moz suggested in [this comment](https://github.com/mozilla-mobile/fenix/issues/8172#issuecomment-584437588) I also added a `E_Fennec_To_Fenix_Migrated` Leanplum event which will be sent when the user has successfully migrated from Fennec to Fenix.

**Comment by st3fan** [comment]
Added @jonalmeida and @pocmo as reviewers

File: app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt [hunk 1]
```diff
@@ -94,7 +95,8 @@ class LeanplumMetricsService(private val application: Application) : MetricsServ
             "klar_installed" to installedApps.contains(MozillaProductDetector.MozillaProducts.KLAR.productName),
             "fxa_signed_in" to application.settings().fxaSignedIn,
             "fxa_has_synced_items" to application.settings().fxaHasSyncedItems,
-            "search_widget_installed" to application.settings().searchWidgetInstalled
+            "search_widget_installed" to application.settings().searchWidgetInstalled,
+            "Fenix" to true
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt]: nit: Not sure if it matters but all other keys here seem to be lowercase.

File: app/src/migration/java/org/mozilla/fenix/MigrationTelemetryListener.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    @UseExperimental(ExperimentalCoroutinesApi::class)
+    fun start() {
+        // Observe for migration completed.
+        store.flowScoped { flow ->
+            flow.collect { state ->
+                Logger("MigrationTelemetryListener").debug("Migration state: ${state.progress}")
+                if (state.progress == MigrationProgress.COMPLETED) {
+                    metrics.track(Event.FennecToFenixMigrated)
+                }
+            }
```
Comment (pocmo) [inline app/src/migration/java/org/mozilla/fenix/MigrationTelemetryListener.kt]: This should work. As an alternative we could also emit a `Fact` from the component. With that you wouldn't need to listen here indefinitely.

File: app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt (hunk 1 above)
Comment (st3fan) [reply app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt]: This is my fault. I asked for a `Fenix` attribute. If this has not landed yet, then lets make it consistent please.

**Comment by st3fan** [comment]
@ValentinTimisica @boek if we want this to land in the last Migration Nightly build (which we will do tomorrow) then this needs a data review and land today.

File: app/src/migration/java/org/mozilla/fenix/MigrationTelemetryListener.kt (hunk 2 above)
Comment (boek) [reply app/src/migration/java/org/mozilla/fenix/MigrationTelemetryListener.kt]: @pocmo if that is easy let's do it. I'm going to land this for now. I filed https://github.com/mozilla-mobile/android-components/issues/5929, if that lands I'll factor out this code and use the Fact instead 👍

Comment (jonalmeida) [reply app/src/migration/java/org/mozilla/fenix/MigrationTelemetryListener.kt]: I've put up a PR for this: https://github.com/mozilla-mobile/android-components/pull/5932

**Pull request merged**